akka-actor-typed/src/main/scala/akka/actor/typed/internal/delivery/WorkPullingProducerController.scala

```diff
@@ -0,0 +1,568 @@
+/*
+ * Copyright (C) 2019-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal.delivery
+
+import java.util.UUID
+import java.util.concurrent.ThreadLocalRandom
+
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+import scala.util.Failure
+import scala.util.Success
+
+import akka.Done
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.receptionist.Receptionist
+import akka.actor.typed.receptionist.ServiceKey
+import akka.actor.typed.scaladsl.ActorContext
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.LoggerOps
+import akka.actor.typed.scaladsl.StashBuffer
+import akka.util.Timeout
+import akka.actor.typed.internal.delivery.DurableProducerQueue.ConfirmationQualifier
+
+object WorkPullingProducerController {
```

Comment:
Maybe FanoutProducerController?

Reply:
I think Work Pulling is a rather established pattern name and the purpose is exactly for that and nothing more so I'd prefer to keep WorkPulling in the name.